### PR TITLE
Add kernel version header implementation

### PR DESCRIPTION
**Pull request by alexandruradovici**
This PR adds the `KernelVersion` header implementation defined in https://github.com/tock/tock/pull/2669.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "elf2tab"
-version = "0.7.0-dev"
+version = "0.8.0-dev"
```
Comment (bradjc) [inline Cargo.toml:3]:
```suggestion
version = "0.7.0-dev"
```

Comment (alexandruradovici) [reply Cargo.toml:3]: Shouldn't we increase the version due to this new header?

Comment (bradjc) [reply Cargo.toml:3]: 0.7 is already an increase.

Comment (alexandruradovici) [reply Cargo.toml]: Got it now :)

**Review by hudson-ayers — APPROVED** [review]
I don't see anything in the updated diff that looks like it could interfere with Alistair's work that was merged before this, I think it is good to go

**Review by bradjc — APPROVED** [review]

**Pull request merged**